### PR TITLE
feat(event-search): Organization Event Json endpoint

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events.py [hunk 1]
```diff
@@ -188,3 +189,32 @@ def get(self, request, organization):
...
+
+class OrganizationEventJsonView(OrganizationEventsEndpointBase):
+    # TODO(lb): should permissions be any different here?
+    # required_scope = 'event:read'
+
+    def get(self, request, organization, event_id):
+        # TODO(lb): This endpoint used to handle both snuba and without snuba.
+        # Couldnt' think of a reason why we wanted without snuba
+
+        filter_params = self.get_filter_params(request, organization)
```
Comment (lauryndbrown) [inline src/sentry/api/endpoints/organization_events.py]: Can throw a no projects error. should be handled.

File: src/sentry/api/endpoints/organization_events.py [hunk 2]
```diff
@@ -188,3 +189,32 @@ def get(self, request, organization):
...
+        project_id = filter_params.get('project_id')
+        if not project_id:
+            return Response({'data': []})
+
+        # TODO(lb): this checks both hex event_id or postgres primary key
```
Comment (lauryndbrown) [inline src/sentry/api/endpoints/organization_events.py]: @markstory do you know what is being sent here? Are there any postgres `event.id` stuff going on or can you just send the snuba event hex?

File: src/sentry/api/endpoints/organization_events.py [hunk 3]
```diff
@@ -188,3 +189,32 @@ def get(self, request, organization):
...
         })
+
+
+class OrganizationEventJsonView(OrganizationEventsEndpointBase):
+    # TODO(lb): should permissions be any different here?
+    # required_scope = 'event:read'
+
+    def get(self, request, organization, event_id):
+        # TODO(lb): This endpoint used to handle both snuba and without snuba.
+        # Couldnt' think of a reason why we wanted without snuba
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: There shouldn't be anymore 'no snuba' now.

File: src/sentry/api/endpoints/organization_events.py (hunk 2 above)
Comment (markstory) [reply src/sentry/api/endpoints/organization_events.py]: I would like to send the snuba hex id.

File: src/sentry/api/endpoints/organization_events.py [hunk 4]
```diff
@@ -188,3 +189,32 @@ def get(self, request, organization):
...
+
+        # TODO(lb): this checks both hex event_id or postgres primary key
+        # do we still need to do this?
+        event = SnubaEvent.objects.from_event_id(event_id, project_id)
+
+        if not event:
+            return Response({'detail': 'Event not found'}, status=404)
+
+        # TODO(lb): not sure this is needed
+        # Event.objects.bind_nodes([event], 'data')
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_events.py]: The current JSON view includes all the data in nodestore.

File: src/sentry/api/urls.py [hunk 5]
```diff
@@ -596,6 +596,11 @@
         OrganizationEventsEndpoint.as_view(),
         name='sentry-api-0-organization-events'
     ),
+    url(
+        r'^organizations/(?P<organization_slug>[^\/]+)/events/(?P<event_id>[^\/]+)/json/$',
```
Comment (markstory) [inline src/sentry/api/urls.py]: I think we *might* need a project id here, as eventids in snuba are only guaranteed unique within a project.

Comment (lauryndbrown) [reply src/sentry/api/urls.py]: @markstory I totally agree. I'll add that to the url. That will simplify my code I think.

File: tests/snuba/api/endpoints/test_organization_events.py [hunk 6]
```diff
@@ -844,3 +846,54 @@ def test_no_projects(self):
...
+        )
+
+    def assert_event(self, data):
+        data = json.loads(data.decode('utf-8'))
+        assert data['event_id'] == self.event_id
+        assert data['user']['email'] == self.user.email
+        assert data['datetime'][:19] == self.min_ago
+        assert data['fingerprint'] == self.fingerprint
+
+    def test_simple(self):
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events.py]: Should we have tests for 404 and no-access scenarios too?

**Comment by lauryndbrown** [comment]
@markstory let me know what you think of the reorganization here. I changed it to a project endpoint because that better matched up with the url I was building.

**Review by markstory — APPROVED** [review]
Looks good to me. Having this at the project level makes sense as that is how events are stored as well.

**Pull request merged**